edx/edx-ora2
# Log more helpful error messages for file uploads

**TL;DR -** log frontend file upload errors as an analytics event to start capturing common error conditions.

JIRA: Based on issues in [EDUCATOR-5406](https://openedx.atlassian.net/browse/EDUCATOR-5406)

**What changed?**
- Log an analytics event for file upload errors.

**Developer Checklist**

**Testing Instructions**

1. Go to an ORA with file uploads enabled
2. Coerce an error on upload, suggest uploading file w/ an unsupported title like `tęšt-fïłê.png`
3. Inspect LMS system logs for an analytics event which should contain the two fields below (plus a lot of context info)

```
  "name": "openassessment.upload_file_error",
  "event": "{\"statusText\": \"TypeError: Failed to execute 'setRequestHeader' on 'XMLHttpRequest': String contains non ISO-8859-1 code point.\"}", 
```

**Reviewer Checklist**

Collectively, these should be completed by reviewers of this PR:


FIY: @edx/masters-devs-gta
